### ISSUE TITLE
Type error in types/core.d.ts

### ISSUE THREAD

**Issue by zingmane** [ivanhofer/typesafe-i18n#127]
### Describe the bug

`An index signature parameter type cannot be a union type. Consider using a mapped object type instead. ts(1337)`

![image](https://user-images.githubusercontent.com/148202/143078417-a71e13cc-28ec-41a8-80bc-52bc03732a17.png)

### Reproduction

I'm using TS 4.3.2 in strict mode.
Unfortunately, I don't have a repo available for it right now.

### Config

```json
{
   "baseLocale": "de",
   "adapter": "node",
   "outputPath": "./src",
   "loadLocalesAsync": false,
   "$schema": "https://unpkg.com/typesafe-i18n@2.52.2/schema/typesafe-i18n.json"
}
```

### Additional information

ts-config.json
```json
{
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "jsx": "react",
    "removeComments": true,
    "sourceMap": true,
    "esModuleInterop": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "target": "es6",
    "lib": [
      "es6"
    ],
    "incremental": false,
    "strict": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "resolveJsonModule": true,
    "outDir": "./lib"
  },
  "exclude": [
    "node_modules",
    "lib"
  ]
}
```

**Comment by ivanhofer** [comment]
I also saw that error today.
The `//@ts-ignore` comment gets removed when running `tsc`. I will fix that in the coming days

**Comment by ivanhofer** [comment]
actually it was a different error that I had discovered before.
Anyway both issues should be fixed in version `2.53.2`
